galata/test/jupyterlab/settings.test.ts

```diff
@@ -250,3 +250,42 @@ test('Opening Keyboard Shortcuts settings does not mangle user shortcuts', async
     .count();
   expect(userPanelLines).toBeLessThan(10);
 });
+
+test('Keyboard Shortcuts: overwriting a shortcut can be cancelled', async ({
+  page
+}) => {
+  // Settings are wide, hide the sidebar to increase available space
+  await page.sidebar.close();
+
+  await page.evaluate(async () => {
+    await window.jupyterapp.commands.execute('settingeditor:open', {
+      query: 'Keyboard Shortcuts'
+    });
+  });
+
+  const shortcutsForm = page.locator('.jp-Shortcuts-ShortcutUI');
+  const filterInput = shortcutsForm.locator('jp-search.jp-FilterBox');
+  await filterInput.click();
+  await filterInput.pressSequentially('merge cell below');
```

Comment:
To answer your question, playwright can pierce through open ShadowDOM to reach the input. So it is indeed more convoluted but the following code works:

```suggestion
  await filterInput.locator('input').fill('merge cell below');
```